src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -509,6 +516,19 @@ private void ParseHelper(string[] args)
                     switchKey = switchKey.Substring(1);
                 }
 
+                // If version is in the commandline, don't continue to look at any other parameters
```

Comment:
I wonder how it's going to interact with existing in Windows PowerShell 
`powershell -version 2.0`

Since we are stopping to build Windows powershell, and core edition doesn't have any `-version 2.0` support, it's probably okey.